### PR TITLE
Add --replay-metadata utility command.

### PR DESCRIPTION
**Pull request by pchote**
For @ihptru.  Mod ID and version will be included in the output once #7353 is merged.

Sample output:

```
paul@Mjolnir:OpenRA+replay-metadata *$ mono OpenRA.Utility.exe ra --replay-metadata OpenRA-2014-10-29T080254Z.rep
OpenRA-2014-10-29T080254Z.rep: 
    MapUid: b6a5b18c23831b54a53db89bf752a7cbf7a306f1
    MapTitle: Chaos Canyon
    StartTimeUtc: 2014-10-29 08-02-55
    EndTimeUtc: 2014-10-29 08-16-34
    Players:
        0: 
            ClientIndex: 0
            Name: Ripley`
            IsHuman: True
            IsBot: False
            FactionName: Soviet
            FactionId: soviet
            Color: 82,93,70
            Team: 1
            SpawnPoint: 6
            IsRandomFaction: False
            IsRandomSpawnPoint: False
            Outcome: Won
            OutcomeTimestampUtc: 2014-10-29 08-15-00
        1: 
            ClientIndex: 2
            Name: Paul
            IsHuman: True
            IsBot: False
            FactionName: Soviet
            FactionId: soviet
            Color: 45,95,205
            Team: 1
            SpawnPoint: 2
            IsRandomFaction: False
            IsRandomSpawnPoint: False
            Outcome: Won
            OutcomeTimestampUtc: 2014-10-29 08-15-00
        2: 
            ClientIndex: 3
            Name: Normal AI
            IsHuman: False
            IsBot: True
            FactionName: Allies
            FactionId: allies
            Color: 114,189,94
            Team: 2
            SpawnPoint: 11
            IsRandomFaction: True
            IsRandomSpawnPoint: False
            Outcome: Lost
            OutcomeTimestampUtc: 2014-10-29 08-14-46
        3: 
            ClientIndex: 4
            Name: Rush AI
            IsHuman: False
            IsBot: True
            FactionName: Soviet
            FactionId: soviet
            Color: 22,234,187
            Team: 2
            SpawnPoint: 10
            IsRandomFaction: True
            IsRandomSpawnPoint: False
            Outcome: Lost
            OutcomeTimestampUtc: 2014-10-29 08-15-00
        4: 
            ClientIndex: 5
            Name: Rush AI
            IsHuman: False
            IsBot: True
            FactionName: Soviet
            FactionId: soviet
            Color: 194,141,107
            Team: 2
            SpawnPoint: 8
            IsRandomFaction: True
            IsRandomSpawnPoint: False
            Outcome: Lost
            OutcomeTimestampUtc: 2014-10-29 08-12-18
        5: 
            ClientIndex: 6
            Name: Normal AI
            IsHuman: False
            IsBot: True
            FactionName: Allies
            FactionId: allies
            Color: 127,182,136
            Team: 2
            SpawnPoint: 9
            IsRandomFaction: True
            IsRandomSpawnPoint: False
            Outcome: Lost
            OutcomeTimestampUtc: 2014-10-29 08-13-23
```

### CODE REVIEW

**Comment by ihptru** [comment]
:+1: , possible error value would be great too

File: OpenRA.Mods.Common/UtilityCommands/ReplayMetadataCommand.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+using System;
+using OpenRA.FileFormats;
+
+namespace OpenRA.Mods.Common.UtilityCommands
+{
+	class ReplayMetadataCommand : IUtilityCommand
+	{
+		public string Name { get { return "--replay-metadata"; } }
+
+		[Desc("Print the game metadata from a replay file")]
```
Comment (Mailaender) [inline OpenRA.Mods.Common/UtilityCommands/ReplayMetadataCommand.cs]:
```c#
[Desc("REPLAYFILE", "Print the game metadata from a replay file.")]
```

**Comment by obrakmann** [comment]
Works well. :+1: when the Desc string has been corrected as mentioned above.

**Comment by pchote** [comment]
Updated and rebased.

**Comment by Mailaender** [comment]
It will crash with replays from the current stable release:

```
mono --debug OpenRA.Utility.exe ra --replay-metadata /home/matthias/.openra/Replays/ra/release-20141029/OpenRA-2014-12-31T140153Z.rep
Error: Utility application crashed. See utility.log for details

Unhandled Exception:
System.NullReferenceException: Object reference not set to an instance of an object
  at OpenRA.Mods.Common.UtilityCommands.ReplayMetadataCommand.Run (OpenRA.ModData modData, System.String[] args) [0x0000a] in /home/matthias/Entwicklung/OpenRA/OpenRA.Mods.Common/UtilityCommands/ReplayMetadataCommand.cs:24 
  at OpenRA.Utility.Program.Main (System.String[] args) [0x0015a] in /home/matthias/Entwicklung/OpenRA/OpenRA.Utility/Program.cs:54 
[ERROR] FATAL UNHANDLED EXCEPTION: System.NullReferenceException: Object reference not set to an instance of an object
  at OpenRA.Mods.Common.UtilityCommands.ReplayMetadataCommand.Run (OpenRA.ModData modData, System.String[] args) [0x0000a] in /home/matthias/Entwicklung/OpenRA/OpenRA.Mods.Common/UtilityCommands/ReplayMetadataCommand.cs:24 
  at OpenRA.Utility.Program.Main (System.String[] args) [0x0015a] in /home/matthias/Entwicklung/OpenRA/OpenRA.Utility/Program.cs:54
```

but works fine with ones recorded with current bleed:

```
mono --debug OpenRA.Utility.exe ra --replay-metadata /home/matthias/.openra/Replays/ra/{DEV_VERSION}/OpenRA-2014-12-31T103637Z.rep
/home/matthias/.openra/Replays/ra/{DEV_VERSION}/OpenRA-2014-12-31T103637Z.rep: 
        MapUid: d7f4992eafbb54d388fb44d5507ea4757e80f0ff
        MapTitle: Singles
        StartTimeUtc: 2014-12-31 11-36-38
        EndTimeUtc: 2014-12-31 11-37-00
        Players:
                0: 
                        ClientIndex: 0
                        Name: Newbie
                        IsHuman: True
                        IsBot: False
                        FactionName: Germany
                        FactionId: germany
                        Color: 75,255,180
                        Team: 0
                        SpawnPoint: 1
                        IsRandomFaction: True
                        IsRandomSpawnPoint: True
                        Outcome: Lost
                        OutcomeTimestampUtc: 2014-12-31 11-36-43
                1: 
                        ClientIndex: 1
                        Name: Rush AI
                        IsHuman: False
                        IsBot: True
                        FactionName: Ukraine
                        FactionId: ukraine
                        Color: 241,107,53
                        Team: 0
                        SpawnPoint: 2
                        IsRandomFaction: True
                        IsRandomSpawnPoint: True
                        Outcome: Won
                        OutcomeTimestampUtc: 2014-12-31 11-36-43
```

**Comment by pchote** [comment]
@Mailaender: That's expected when you parse it a replay file in a format it doesn't understand.  I could eat the exception and do something else, but I will need some feedback from @ihptru how he wants that case to look.  When it throws an exception he can at least check the error code and gracefully fail.

**Comment by Mailaender** [comment]
I thought so. Code looking good :+1: and verified it works. :white_check_mark:

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/56ccc5c87aa71381a3b83be9fb4f1d8cfdcaaee6...75d8d724a5e08b356ef897a3f2a25341eb963833)

**Comment by StexoO** [comment]
I don't know if I'm late to the party but this utility is what I expected the most in this game ! Now a world OpenRA ladder is possible. And maybe one day complete statistics on every player. And maybe another day an "Auto-balance teams" button on the servers. Really good job ! :pray:

**Comment by SoScared** [comment]
:+1: Community plebs approve! And don't forget to claim the small bounty in due time.

**Comment by Mailaender** [comment]
A partly redundant, less machine, more human readable version is being suggested in https://github.com/OpenRA/OpenRA/pull/8465.

**Pull request merged**